nbs/02_match_class.ipynb

```diff
@@ -224,7 +224,9 @@
     "        self.graph: DiGraph = graph\n",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #7.        # Step 1: Call find_matches for each collection path
</pre><p>this has alot of redundancy. The collection patterns should be patterns like the LHS patterns, but with the var of the previous phase fixed as constants.</p><p>Looking for all collections for each match creates too many intermediate results that will be pruned</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/DeanLight/graph_rewrite/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Reply:
<p>We think we understand how to avoid these redundancies - we'll check this new idea, and will talk about it in our meeting to make sure we're on the same page </p>